sdx-submitter.go

```diff
@@ -0,0 +1,168 @@
+	// Use values from environment variables if no value empty
+	getFromEnvIfEmpty(name, "RABBITMQ_DEFAULT_USER", "guest")
+	getFromEnvIfEmpty(password, "RABBITMQ_DEFAULT_PASS", "guest")
+	getFromEnvIfEmpty(port, "RABBITMQ_PORT", "5672")
+	getFromEnvIfEmpty(host, "RABBITMQ_HOST", "rabbit")
+	getFromEnvIfEmpty(queue, "RABBIT_SURVEY_QUEUE", "rabbit")
+	getFromEnvIfEmpty(exchange, "RABBITMQ_EXCHANGE", "rabbit")
+	getFromEnvIfEmpty(vhost, "RABBITMQ_DEFAULT_VHOST", "%2f")
+	if *url == "" {
+		*url = fmt.Sprintf("amqp://%s:%s@%s:%s/%s", *name, *password, *host, *port, *vhost)
+	} else {
+		fmt.Println("Url use overrides specific parameters") //A specific URL is set , so use it
+	}
+}
+
+func getFromEnvIfEmpty(target *string, key string, defaultValue string)  {
+	if *target == "" {
+		if value, present := os.LookupEnv(key); present {
+			*target = value
+		} else {
+			*target = defaultValue
+		}
+	}
+}
+
+// Consider adding StdIn reading here to support piping ?
+func getBody(file_path string)([]byte){
+	var msg_body []byte
+	var err error
+	if len(file_path) != 0 {
+		msg_body, err = ioutil.ReadFile(file_path)
+		failOnError(err, "Cannot read from file")
+	} else {panic("No file name supplied")}  // Only mandatory until we support piping
+	return msg_body
+}
+
+func failOnError(err error, msg string) {
+	if err != nil {
+		panic(fmt.Sprintf("%s: %s", msg, err))
+	}
+}
+
+func sendToRabbit(url string, exchange string, queue string, routing_key string, err error, msg_body []byte) {
+	// Connect to Rabbit
+	conn, err := amqp.Dial(url)
+	failOnError(err, "Failed to connect to RabbitMQ")
+	ch, err := conn.Channel()
+	failOnError(err, "Failed to open a channel")
+
+	defer conn.Close()
+	defer ch.Close()
+
+
+	// REMOVE THIS CODE !!!  Do not Declare in prod - must exist Prior to run ?
+	rabbitPrepare(ch, exchange, queue, routing_key)
+	// END OF CODE TO REMOVE
+
+
+	err = ch.Publish(
+		exchange,    // exchange
+		routing_key, // routing key
+		false,       // mandatory
+		false,       // immediate
+		amqp.Publishing{
+			ContentType: "text/plain",
+			Body: msg_body,
+		})
+
+	failOnError(err, "Failed to publish a message")
+}
+
+
+
+
+// Not for Prod
+func rabbitPrepare(ch *amqp.Channel, exchange string, queue string, routing_key string)  {
+	err := ch.ExchangeDeclare(
+		exchange, // name
+		"topic",
+		true,  // durable
+		false, // auto-deleted
+		false, // internal
+		false, // no-wait
+		nil,   // arguments
+	)
+	failOnError(err, "Failed to declare an exchange")
+	q, err := ch.QueueDeclare(
+		queue, // name
+		false,   // durable
+		false,   // delete when unused
+		false,   // exclusive
+		false,   // no-wait
+		nil,     // arguments
+	)
+	Use(q)
+	failOnError(err, "Could not create queue ")
+	err = ch.QueueBind(
+		queue,       // queue name
+		routing_key, // routing key
+		exchange,    // exchange
+		false,
+		nil)
+	failOnError(err, fmt.Sprintf("could not bind queue %s to exchange %s", queue, exchange))
+}
+
+func Use(vals ...interface{}) {
```

Comment:
I assume this is to avoid the "`x var is unused`" compiler error?

Should either use the var (you can use `q.Name` when you're queue binding) or throw it away explicitly with `_`

Reply:
yep - only meant for dev , but removed